test/unit_test/dsjson_parser_test.cc

```diff
@@ -266,8 +267,9 @@ BOOST_AUTO_TEST_CASE(parse_dsjson_cats_no_label)
   }
 }
 )";
-  auto vw = VW::initialize("--dsjson -t --cats 4 --min_value=185 --max_value=23959 --bandwidth 1 --no_stdin --quiet",
-      nullptr, false, nullptr, nullptr);
+  auto vw = VW::initialize(

```

Comment:
I find this confusing, we need to turn chain hash on for json, even if the feature values are numerical?

Reply:
The alternative is keeping track of if a text feature was or wasn't seen and warn only if one was seen. Given we are migrating to this behavior it seems simpler to just enforce it on an input type level?

Reply:
yeah..